### PR TITLE
Install build tools

### PR DESCRIPTION
**Pull request by synarete**
Use helper script to install auxiliary build tools, using local GOBIN.

### CODE REVIEW

File: hack/install-tools.sh [hunk 1]
```diff
@@ -0,0 +1,80 @@
+#!/bin/sh
+# helper script to install build auxiliary tools in local directory
+#
+# usage:
+#   GOBIN=<dir> [TMPDIR=<tmpdir>] install-tools.sh --<tool-name>
+#
+set -e
+GO_CMD=${GO_CMD:-$(command -v go)}
+GOBIN=${GOBIN:-$(pwd)/bin}
+TMPDIR=${TMPDIR:-$(pwd)/tmp}
```
Comment (phlogistonjohn) [inline hack/install-tools.sh:10]: Let's use `mktmp -d` in the functions that need a temporary dir.

File: Makefile [hunk 2]
```diff
@@ -52,6 +52,10 @@ endif
 # Get current GOARCH
 GOARCH?=$(shell $(GO_CMD) env GOARCH)
 
+# Local (alternative) GOBIN for auxiliary build tools
+GOBIN_ALT:=$(CURDIR)/build/tools/bin/
```
Comment (phlogistonjohn) [inline Makefile:56]: How about just `$(CURDIR)/.bin`  ?

File: Makefile [hunk 3]
```diff
@@ -163,24 +167,25 @@ container-push:
 # download controller-gen if necessary
 controller-gen:
 ifeq (, $(shell command -v controller-gen ;))
-	@echo "controller-gen not found in PATH, checking in GOBIN ($(GOBIN))..."
-ifeq (, $(shell command -v $(GOBIN)/controller-gen ;))
-	$(GO_CMD) install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.6.2
-	@echo "controller-gen installed in GOBIN"
+	@echo "controller-gen not found in PATH, checking $(GOBIN_ALT)"
+ifeq (, $(shell command -v $(GOBIN_ALT)/controller-gen ;))
+	@GOBIN=$(GOBIN_ALT) GO_CMD=$(GO_CMD) hack/install-tools.sh --controller-gen
```
Comment (phlogistonjohn) [inline Makefile]: Just to be on the safe side: `$(CURDIR)/hack/install-tools.sh` ?

**Review by phlogistonjohn — CHANGES_REQUESTED** [review]
Really great start thanks! It's a lot simpler than than what I tried a few weeks ago and that makes it a lot more maintainable.

File: hack/install-tools.sh (hunk 1 above)
Comment (synarete) [reply hack/install-tools.sh:10]: ACK

File: Makefile (hunk 2 above)
Comment (synarete) [reply Makefile:56]: ACK

File: Makefile (hunk 3 above)
Comment (synarete) [reply Makefile]: ACK

Comment (synarete) [reply Makefile]: (and we can even do it better ;)

**Review by phlogistonjohn — APPROVED** [review]
Very nice, thanks!

**Comment by phlogistonjohn** [comment]
I'm leaving it open for a couple of days to give @raghavendra-talur a chance to look as we've discussed this topic regarding other projects in the past.

**Review by raghavendra-talur — APPROVED** [review]

**Pull request merged**